### ISSUE TITLE
`/api/v1/query` doesn't support matrix type anymore

### ISSUE THREAD

**Issue by juliusv** [prometheus/prometheus#846]
Try:

http://localhost:9090/api/v1/query?query=up%5B1m%5D&time=1435229840.642

Response:

```json
{"status":"error","errorType":"bad_data","error":"invalid expression type \"matrix\" for range query, must be scalar or vector"}
```

Expected: valid matrix response.

**Comment by juliusv** [comment]
@fabxc

**Comment by fabxc** [comment]
Fixed in #848
